sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -171,8 +172,7 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
             location = fileIndex,
             partitionSchema = partitionSchema,
             dataSchema = dataSchema,
-            // We don't support hive bucketed tables, only ones we write out.
-            bucketSpec = None,
+            bucketSpec = bucketSpec,
```

Comment:
here we are reading hive table as a data source table, so the scan operator is actually `FileSourceScanExec`, which recognizes bucketing.